clinica/utils/filemanip.py

```diff
@@ -1,209 +1,476 @@
     split = [image_id.split("_") for image_id in id_bids_or_caps_files]
     subject_ids = [p_id[0] for p_id in split]
     session_ids = [s_id[1] for s_id in split]
     return subject_ids, session_ids
 
 
-def extract_crash_files_from_log_file(filename):
-    """Extract crash files (*.pklz) from `filename`."""
-    import os
-    import re
+def extract_crash_files_from_log_file(filename: str) -> List[str]:
+    """Extract crash files (*.pklz) from `filename`.
 
-    assert os.path.isfile(
-        filename
-    ), f"extract_crash_files_from_log_file: filename parameter is not a file ({filename})"
+    Parameters
+    ----------
+    filename: str
+        Path to log file for which the crash files should be extracted.
 
-    log_file = open(filename, "r")
+    Returns
+    -------
+    crash_files: List[str]
+        List of crash files.
+    """
+    import re
+    from pathlib import Path
+
+    filename = Path(filename)
+    if not filename.is_file():
+        raise ValueError(
+            f"extract_crash_files_from_log_file: filename parameter is not a file ({filename})"
+        )
     crash_files = []
-    for line in log_file:
-        if re.match("(.*)crashfile:(.*)", line):
-            crash_files.append(line.replace("\t crashfile:", "").replace("\n", ""))
+    with open(filename, "r") as log_file:
+        for line in log_file:
+            if re.match("(.*)crashfile:(.*)", line):
+                crash_files.append(line.replace("\t crashfile:", "").replace("\n", ""))
 
     return crash_files
 
 
-def read_participant_tsv(tsv_file):
+def read_participant_tsv(tsv_file: str) -> Tuple[List[str], List[str]]:
     """Extract participant IDs and session IDs from TSV file.
 
-    Raise:
-        ClinicaException if tsv_file is not a file
-        ClinicaException if participant_id or session_id column is missing from TSV file
+    Parameters
+    ----------
+    tsv_file: str
+        Participant TSV file from which to extract the participant and session IDs.
+
+    Returns
+    -------
+    participants: List[str]
+        List of participant IDs.
+
+    sessions: List[str]
+        List of session IDs.
+
+    Raises
+    ------
+    ClinicaException
+        If `tsv_file` is not a file.
+        If `participant_id` or `session_id` column is missing from TSV file.
+
+    Examples
+    --------
+    >>> dframe = pd.DataFrame({
+    ...     "participant_id": ["sub-01", "sub-01", "sub-02"],
+    ...     "session_id": ["ses-M000", "ses-M006", "ses-M000"],
+    ...})
+    >>> dframe.to_csv("participants.tsv", sep="\t")
+    >>> read_participant_tsv("participant.tsv")
+    (["sub-01", "sub-01", "sub-02"], ["ses-M000", "ses-M006", "ses-M000"])
     """
-    import os
+    from pathlib import Path
 
     import pandas as pd
 
     from clinica.utils.exceptions import ClinicaException
 
-    if not os.path.isfile(tsv_file):
+    tsv_file = Path(tsv_file)
+    if not tsv_file.is_file():
         raise ClinicaException(
```

Comment:
It is more Pythonic to do:

```python
try:
    df = pd.read_csv(...)
except ... :
     ...
```

than testing the file.

Actually, the file could exist without permissions to read it.

Reply:
I agree => implemented in [e263a47](https://github.com/aramis-lab/clinica/pull/781/commits/e263a475637ffb76b31a835d72afe2c1977b5b2b)